src/state/operator/selectors.js

```diff
@@ -46,28 +45,53 @@ const totalAvailable = ( { load, capacity } ) => ( capacity - defaultTo( 0, load
 const RESERVE_MAX_CHATS = 2;
 
 /**
-/* Compare function for sorting operator priority. 
- *
- * The first operator in the resulting list should be the next one
- * to be assigned to a chat. This is used for balancing chat load
- * evenly across operators.
+ * Comparison function for sorting operators by the requestingChat property
  *
  * @param {Operator} a First operator to compare
  * @param {Operator} b Second operator to compare
- * @return {Number} -1 if operator a should come first
- *                  0 if both operators are equal priority
- *                  1 if operator b should come first
+ * @return {Number} -1 if operator a is requesting a chat
+ *                  0 if neither are requesting a chat
+ *                  1 if operator b is requesting a chat
  */
-const compareOperatorPriority = ( a, b ) => {
-	// When comparing two operators, always prioritise the one whose
-	// status is available over the one who is in reserve
-	if ( a.status === STATUS_RESERVE && b.status === STATUS_AVAILABLE ) {
-		return 1;
+const compareByRequestForChat = ( a, b ) => {
+	// Neither operators are requesting a chat so don't alter their order
+	if ( ! a.requestingChat && ! b.requestingChat ) {
+		return 0;
 	}
-	if ( a.status === STATUS_AVAILABLE && b.status === STATUS_RESERVE ) {
-		return -1;
+
+	// Only one of the operators is requesting a chat
+	if ( a.requestingChat !== b.requestingChat ) {
+		return a.requestingChat ? -1 : 1;
 	}
 
+	// Both are requesting a chat so lets use their status
+	// To break the tie
+	const statusComparison = compareOperatorByStatus( a, b );
+	if ( statusComparison !== 0 ) {
+		return statusComparison;
+	}
+
+	// They're still tied so lets try to break it using
+	// their chat load
+	const loadComparison = compareOperatorsByChatLoad( a, b );
+	if ( loadComparison !== 0 ) {
+		return loadComparison;
+	}
```

Comment:
Instead of repeating the two above comparisons, we could just return 0 and let `compareOperatorPriority` do the other calcs. In fact I think this function could be reduced to just:

```js
const compareByRequestForChat = ( a, b ) => {
	if ( a.requestingChat !== b.requestingChat ) {
		return a.requestingChat ? -1 : 1;
	}

	return 0;
}
```

If neither or both of the operators are requesting an extra chat, then we don't have any useful information to determine their priority against each other.